src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -549,15 +549,32 @@ static void RemoveOtherSupportsOnDifferentPlatforms(SmallDictionary<string, Plat
             {
                 symbol = GetAccessorMethod(platformSpecificMembers, symbol, GetPropertyAccessors(property, operation));
             }
-
-            if (symbol is IEventSymbol iEvent)
+            else if (symbol is IEventSymbol iEvent)
             {
                 var accessor = GetEventAccessor(iEvent, operation);
                 if (accessor != null)
                 {
                     symbol = accessor;
                 }
             }
+            else if (symbol is IMethodSymbol method)
+            {
+                if (method.IsGenericMethod && TryGetSymbol(platformSpecificMembers, out var foundSymbol, method.TypeArguments))
+                {
+                    symbol = foundSymbol;
```

Comment:
For this case, I feel we should analyze both the original method symbol and also the found type symbol (or in fact all type symbols that have platform attributes). The current implementation means that we will not validate the platform attributes on the method itself if we find a type argument with platform attributes.

Reply:
This section is for ReportDiagnostics(...), i.e. the analysis already done and reporting diagnostic, here just trying load the correct symbol for the message, the analysis section done here https://github.com/dotnet/roslyn-analyzers/pull/4403/files#r517676304